api/crush_rule.proto

```diff
@@ -0,0 +1,59 @@
+syntax = "proto3";  
+
+import "google/protobuf/empty.proto";
+
+option go_package = "github.com/seaphony/ceph-api/api/ceph;pb";
+
+package ceph;
+
+service CrushRule {
+  rpc CreateRule (CreateRuleRequest) returns (google.protobuf.Empty) {}
+  rpc DeleteRule (DeleteRuleRequest) returns (google.protobuf.Empty) {}
+  rpc GetRule (GetRuleRequest) returns (GetRuleResponse) {}
+  rpc ListRules (google.protobuf.Empty) returns (ListRulesResponse) {}
+}
+
+
+message Rule {
+    int64 rule_id = 1;       
+    string rule_name = 2;    
+    int64 ruleset = 3;       
+    int64 type = 4;          
+    int64 min_size = 5;      
+    int64 max_size = 6;      
+    repeated Step steps = 7; 
+}
+
+message Step {
+    map<string, string> entries = 1; 
+}
+
+// CREATE RULE
+message CreateRuleRequest {
+    string device_class = 1;
+    string failure_domain = 2;
+    string name = 3;
+    string pool_type = 4;
```

Comment:
consider using [enum](https://protobuf.dev/programming-guides/enum/) (replication, erasure) here. In this case it will be more clear which values are supported and what is default value.